pkg/controller/csr_check.go

```
@@ -554,22 +585,30 @@ func getServingCert(c client.Client, nodeName string, ca *x509.CertPool) (*x509.
+		},
 	}
 
 	klog.Infof("retrieving serving cert from %s (%s)", nodeName, kubelet)
 
-	conn, err := tls.DialWithDialer(dialer, "tcp", kubelet, tlsConfig)
+	conn, err := dialer.DialContext(ctx, "tcp", kubelet)
 	if err != nil {
 		return nil, err
 	}
 
 	defer conn.Close()
```

golangci-lint (2.12.2)

[error] Error return value of `conn.Close` is not checked

(errcheck)